### ISSUE TITLE
Typed Events

### ISSUE THREAD

**Issue by masseelch** [aleksanderwozniak/table_calendar#208]
I think it would be a good idea to have the events / holidays typed with a TypeVariable. 

Example:
```
class TableCalendar<T> extends StatefulWidget {
  [...]
  final Map<DateTime, List<T>> events;
  final Map<DateTime, List<T>> holidays;
  [...]
}
```

If you agree i am happy to contribute.

**Comment by aleksanderwozniak** [comment]
That's a nice idea - feel free to go for it.

**Comment by masseelch** [comment]
What branch to form from?

**Comment by aleksanderwozniak** [comment]
You can use master for #200 since it is not a breaking change, and use dev for this one.

**Comment by masseelch** [comment]
Fixed in #210
